content-security-policy/webrtc/webrtc.js

```diff
@@ -0,0 +1,62 @@
+
+// Creates two RTCPeerConnection and tries to connect them. Returns
+// "allowed" if the connection is permitted, "blocked" if it is
+// blocked on both sides and "inconsistent" in the even that the
+// result is the same on both sides (should never happen).
+async function tryConnect() {
+    const iceServers = [{urls: "stun:stun.l.google.com:19302"}];
+    const pc1 = new RTCPeerConnection({iceServers});
+    const pc2 = new RTCPeerConnection({iceServers});
+
+    // Returns a promise which resolves to a boolean which is true
+    // if and only if pc.iceConnectionState settles in the "failed"
+    // state, and never transitions to any state other than "new"
+    // or "failed."
+    const pcFailed = (pc) => {
+        pc.onicecandidate = ({candidate}) => {
+            pc.addIceCandidate(candidate);
+        };
+        return new Promise((resolve, _reject) => {
+            pc.onicegatheringstatechange = (e) => {
+                if(pc.iceGatheringState === "complete") {
+                    resolve(pc.iceConnectionState === "failed");
+                } else if(pc.iceConnectionState !== "new") {
+                    resolve(false);
+                }
+            };
```

Comment:
One more issue:

While it's correct to look at iceConnectionState, I think we need to listen to `pc.oniceconnectionstatechange` here, not `pc.onicegatheringstatechange`, oherwise iceConnectionState is not guaranteed to have been updated yet.

The timing between updates to the two states isn't super-clear in the spec (but I believe it is specified): Most of the time, the two states are timing independent, but for [failed](https://w3c.github.io/webrtc-pc/#dom-rtcicetransportstate-failed), iceConnectionState is causally dependent on multiple things, one of them being _"finished gathering"_, gatheringState is updated ([often way](https://github.com/w3c/webrtc-pc/issues/2698#issuecomment-987443817)) ahead of iceConnectionState.

If we want to test that the PAC timer is bypassed on CSP block (which it should be), we'd need to test that differently somehow, e.g. timeout after 1 second.

Reply:
Done.